qa/vagrant/src/main/java/org/elasticsearch/packaging/test/WindowsServiceTestCase.java

```diff
@@ -0,0 +1,209 @@
+
+    public void test30StartStop() throws IOException {
+        sh.run(serviceScript + " install");
+        sh.run(serviceScript + " start");
+        ServerUtils.waitForElasticsearch();
+        sh.run(serviceScript + " stop");
+        sh.runIgnoreExitCode("Wait-Process -Name \"elasticsearch-service-x64\" -Timeout 10");
+        sh.run(serviceScript + " remove");
+    }
+
+    public void test31StartNotInstalled() throws IOException {
+        Result result = sh.runIgnoreExitCode(serviceScript + " start");
+        assertThat(result.stdout, result.exitCode, equalTo(1));
+        assertThat(result.stdout, containsString("Failed starting '" + DEFAULT_ID + "' service"));
+    }
+
+    public void test32StopNotStarted() throws IOException {
+        sh.run(serviceScript + " install");
+        Result result = sh.run(serviceScript + " stop"); // stop is ok when not started
+        assertThat(result.stdout, containsString("The service '" + DEFAULT_ID + "' has been stopped"));
+    }
+
+    /*
+    // TODO: need to make JAVA_HOME resolve at install time for this to work
+    // see https://github.com/elastic/elasticsearch/issues/23097
+    public void test33JavaChanged() throws IOException {
+        sh.run(serviceScript + " install");
+        runWithoutJava(serviceScript + "start");
+        ServerUtils.waitForElasticsearch();
+        sh.run(serviceScript + " stop");
+        sh.runIgnoreExitCode("Wait-Process -Name \"elasticsearch-service-x64\" -Timeout 10");
+        sh.run(serviceScript + " remove");
+    }*/
+
+    public void test60Manager() throws IOException {
+        Path serviceMgr = installation.bin("elasticsearch-service-mgr.exe");
+        Path tmpServiceMgr = serviceMgr.getParent().resolve(serviceMgr.getFileName() + ".tmp");
+        Files.move(serviceMgr, tmpServiceMgr);
+        Path fakeServiceMgr = serviceMgr.getParent().resolve("elasticsearch-service-mgr.bat");
+        Files.write(fakeServiceMgr, Arrays.asList("echo \"Fake Service Manager GUI\""));
+        Shell sh = new Shell();
+        Result result =  sh.run(serviceScript + " manager");
+        TestCase.assertTrue(result.stdout, result.stdout.contains("Fake Service Manager GUI"));
+
+        // check failure too
+        Files.write(fakeServiceMgr, Arrays.asList("echo \"Fake Service Manager GUI Failure\"", "exit 1"));
+        result =  sh.runIgnoreExitCode(serviceScript + " manager");
+        TestCase.assertEquals(1, result.exitCode);
+        TestCase.assertTrue(result.stdout, result.stdout.contains("Fake Service Manager GUI Failure"));
+        Files.move(tmpServiceMgr, serviceMgr);
+    }
+
+    public void test70UnknownCommand() {
+        Result result = sh.runIgnoreExitCode(serviceScript + " bogus");
+        assertThat(result.exitCode, equalTo(1));
+        assertThat(result.stdout, containsString("Unknown option \"bogus\""));
+    }
+
+    private void assertService(String id, String status, String displayName) {
```

Comment:
Nit: can you put this and `runWithoutJava` next to each other

Reply:
done